Sources/SwiftFixIt/SwiftFixit.swift

```diff
@@ -0,0 +1,409 @@
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import struct Basics.AbsolutePath
+import protocol Basics.FileSystem
+import var Basics.localFileSystem
+import struct Basics.SwiftVersion
+
+import struct SwiftDiagnostics.Diagnostic
+import struct SwiftDiagnostics.DiagnosticCategory
+import protocol SwiftDiagnostics.DiagnosticMessage
+import enum SwiftDiagnostics.DiagnosticSeverity
+import struct SwiftDiagnostics.DiagnosticsFormatter
+import struct SwiftDiagnostics.FixIt
+import protocol SwiftDiagnostics.FixItMessage
+import struct SwiftDiagnostics.GroupedDiagnostics
+import struct SwiftDiagnostics.MessageID
+
+@_spi(FixItApplier)
+import enum SwiftIDEUtils.FixItApplier
+
+import struct SwiftParser.Parser
+
+import struct SwiftSyntax.AbsolutePosition
+import struct SwiftSyntax.SourceFileSyntax
+import class SwiftSyntax.SourceLocationConverter
+import struct SwiftSyntax.Syntax
+
+import struct TSCBasic.ByteString
+import struct TSCUtility.SerializedDiagnostics
+
+private enum Error: Swift.Error {
+    case unexpectedDiagnosticSeverity
+    case failedToResolveSourceLocation
+}
+
+// FIXME: An interface for tests to work around private initializers in `TSCUtility.SerializedDiagnostics`.
+protocol AnySourceLocation {
+    var filename: String { get }
+    var line: UInt64 { get }
+    var column: UInt64 { get }
+    var offset: UInt64 { get }
+}
+
+// FIXME: An interface for tests to work around private initializers in `TSCUtility.SerializedDiagnostics`.
+protocol AnyFixIt {
+    associatedtype SourceLocation: AnySourceLocation
+
+    var start: SourceLocation { get }
+    var end: SourceLocation { get }
+    var text: String { get }
+}
+
+// FIXME: An interface for tests to work around private initializers in `TSCUtility.SerializedDiagnostics`.
+protocol AnyDiagnostic {
+    associatedtype SourceLocation: AnySourceLocation
+    associatedtype FixIt: AnyFixIt where FixIt.SourceLocation == SourceLocation
+
+    var text: String { get }
+    var level: SerializedDiagnostics.Diagnostic.Level { get }
+    var location: SourceLocation? { get }
+    var category: String? { get }
+    var categoryURL: String? { get }
+    var flag: String? { get }
+    var ranges: [(SourceLocation, SourceLocation)] { get }
+    var fixIts: [FixIt] { get }
+}
+
+extension SerializedDiagnostics.Diagnostic: AnyDiagnostic {}
+extension SerializedDiagnostics.SourceLocation: AnySourceLocation {}
+extension SerializedDiagnostics.FixIt: AnyFixIt {}
+
+/// The backing API for `SwiftFixitCommand`.
+package struct SwiftFixIt: ~Copyable {
```

Comment:
`~Copyable` might not work with self-hosted CI.

Reply:
Commented them out. Helpful to uncomment at desk.